### ISSUE TITLE
[feature request] logits processor perfomance issue

### ISSUE THREAD

**Issue by akhoroshev** [NVIDIA/TensorRT-LLM#1681]
Current arch of tensorrtllm means that logits processor (for both [executor](https://github.com/NVIDIA/TensorRT-LLM/blob/main/cpp/include/tensorrt_llm/executor/types.h#L55) and [batch_manager](https://github.com/NVIDIA/TensorRT-LLM/blob/main/cpp/include/tensorrt_llm/batch_manager/llmRequest.h#L58)) will be called independently for **each request**.

But it is bad approach in terms of performance.

For example if i have throughput 5000token/s without custom logits processors then that is equivalent to the system generating each token every 200 µs.

But if i add custom logit processor for each request, it increases latency of each token generation on about 15-20 µs (it depends on complexity of logits processor).

**It leads to performance drop of all system on about 10%.**

This can be avoided by allowing **batch processing** of logits for queries.

**Comment by MartinMarciniszyn** [comment]
A batched logits post processor has been released in the [last update](https://github.com/NVIDIA/TensorRT-LLM/pull/1793) of the `main` branch
